### PR TITLE
Use keras_graph when obtaining gradients

### PR DESCRIPTION
**Pull request by danaugrs**
See https://github.com/tensorflow/tensorflow/issues/26098.

### CODE REVIEW

**Review by alextp — DISMISSED** [review]

**Comment by robieta** [comment]
I'm a little worried about always dropping into graph mode. (I'll respond in more depth in #26098)

File: tensorflow/python/ops/gradients_util.py [hunk 1]
```diff
@@ -542,9 +542,6 @@ def _GradientsHelper(ys,
                      unconnected_gradients=UnconnectedGradients.NONE,
                      src_graph=None):
   """Implementation of gradients()."""
-  if context.executing_eagerly():
```
Comment (robieta) [inline tensorflow/python/ops/gradients_util.py]: I think this still applies. This PR makes V2 optimizers step into graph mode to get updates (and `context.executing_eagerly()` will return False when in that scope), but the restriction is still present in general.

File: tensorflow/python/keras/optimizer_v2/optimizer_v2.py [hunk 2]
```diff
@@ -445,7 +445,8 @@ def apply_grad_to_update_var(var, grad):
       return apply_updates
 
   def get_updates(self, loss, params):
-    grads = self.get_gradients(loss, params)
+    with backend.get_graph().as_default():
```
Comment (robieta) [inline tensorflow/python/keras/optimizer_v2/optimizer_v2.py]: I think this should be in get_gradients rather than get_updates.

**Review by robieta — CHANGES_REQUESTED** [review]
In general I think this is quite reasonable. If we're under a graph context then `backend.get_graph` will grab that (so it's safe even outside the Keras framework).

File: tensorflow/python/ops/gradients_util.py (hunk 1 above)
Comment (danaugrs) [reply tensorflow/python/ops/gradients_util.py]: Yeah I agree - reinstated the check.

File: tensorflow/python/keras/optimizer_v2/optimizer_v2.py (hunk 2 above)
Comment (danaugrs) [reply tensorflow/python/keras/optimizer_v2/optimizer_v2.py]: Moved it.

**Review by robieta — APPROVED** [review]
